AntonioLok/online-shopping-reactJS-Backend
# feat: add precommit hooks to check for commit message

# Problems
1- During my last commit, a warning was reported:
![image](https://user-images.githubusercontent.com/14690224/66720614-076a5b00-edcd-11e9-95ef-0c01935bac39.png)
This PR addresses that warning by moving the precommit script to a `.huskyrc`.

2 - Add pre-commit script to check for commit message. The one i've set up follows the Conventional Commit specifications https://www.conventionalcommits.org/en/v1.0.0/.

# Test
- Install latest dependencies with `npm i`
- Make any changes on any file, ie. `middlewares/get-token-from-header.js`
- Then attempt to commit the changes. You will notice that the precommit hooks are getting triggered. The commit message should be structured as follows: `<type>[optional scope]: <description>` where type must be one of [build, chore, ci, docs, feat, fix, perf, refactor, revert, style, test]. If the commit message is not structure as mentioned, the commit will be aborted.
